src/main/java/com/amazonaws/services/kinesis/samples/stocktrades/processor/StockTradeRecordProcessor.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Amazon Software License (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/asl/
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.services.kinesis.samples.stocktrades.processor;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import software.amazon.kinesis.exceptions.InvalidStateException;
+import software.amazon.kinesis.exceptions.ShutdownException;
+import software.amazon.kinesis.exceptions.ThrottlingException;
+import software.amazon.kinesis.lifecycle.events.InitializationInput;
+import software.amazon.kinesis.lifecycle.events.LeaseLostInput;
+import software.amazon.kinesis.processor.RecordProcessorCheckpointer;
+import software.amazon.kinesis.lifecycle.events.ProcessRecordsInput;
+import software.amazon.kinesis.lifecycle.events.ShardEndedInput;
+import software.amazon.kinesis.lifecycle.events.ShutdownRequestedInput;
+import software.amazon.kinesis.processor.ShardRecordProcessor;
+
+import com.amazonaws.services.kinesis.samples.stocktrades.model.StockTrade;
+import software.amazon.kinesis.retrieval.KinesisClientRecord;
+
+/**
+ * Processes records retrieved from stock trades stream.
+ *
+ */
+public class StockTradeRecordProcessor implements ShardRecordProcessor {
+
+    private static final Log log = LogFactory.getLog(StockTradeRecordProcessor.class);
+
+    private String kinesisShardId;
+
+    // Reporting interval
+    private static final long REPORTING_INTERVAL_MILLIS = 60000L; // 1 minute
+    private long nextReportingTimeInMillis;
+
+    // Checkpointing interval
+    private static final long CHECKPOINT_INTERVAL_MILLIS = 60000L; // 1 minute
+    private long nextCheckpointTimeInMillis;
+
+    // Aggregates stats for stock trades
+    private StockStats stockStats = new StockStats();
+
+    @Override
+    public void initialize(InitializationInput initializationInput) {
+        kinesisShardId = initializationInput.shardId();
+        log.info("Initializing record processor for shard: " + kinesisShardId);
+        log.info("Initializing @ Sequence: " + initializationInput.extendedSequenceNumber().toString());
+
+        nextReportingTimeInMillis = System.currentTimeMillis() + REPORTING_INTERVAL_MILLIS;
+        nextCheckpointTimeInMillis = System.currentTimeMillis() + CHECKPOINT_INTERVAL_MILLIS;
+    }
+
+    @Override
+    public void processRecords(ProcessRecordsInput processRecordsInput) {
+        // TODO: Implement method
```

Comment:
Are these TODOs intentional?

Reply:
yeah the missing code is provided in the tutorial.